Adapted from f5e61804d7a8ad6afc4aa4484e2c05f24ddb396b Mon Sep 17 00:00:00 2001
From: Amar Takhar <verm@darkbeer.org>
Date: Wed, 15 Jan 2014 21:44:48 +0000
Subject: Fixes for FreeBSD 10.0 + clang

Signed-off-by: Raymond Wagner <rwagner@mythtv.org>
(cherry picked from commit 366b73bc672d45be94b981a944f9f415b44ea13c)
---
diff --git mythtv/libs/libmyth/remoteutil.h mythtv/libs/libmyth/remoteutil.h
index 7acff6a..4dc6a54 100644
--- mythtv/libs/libmyth/remoteutil.h
+++ mythtv/libs/libmyth/remoteutil.h
@@ -1,6 +1,8 @@
 #ifndef REMOTEUTIL_H_
 #define REMOTEUTIL_H_
 
+#include <time.h>
+
 #include <QStringList>
 #include <QDateTime>
 
diff --git mythtv/libs/libmythbase/mythmiscutil.h mythtv/libs/libmythbase/mythmiscutil.h
index bfb171e..91dea13 100644
--- mythtv/libs/libmythbase/mythmiscutil.h
+++ mythtv/libs/libmythbase/mythmiscutil.h
@@ -2,6 +2,7 @@
 #define MYTHMISCUTIL_H_
 
 #include <stdint.h>
+#include <time.h>
 
 #include <algorithm>
 using namespace std;
diff --git mythtv/libs/libmythbase/mythtimezone.cpp mythtv/libs/libmythbase/mythtimezone.cpp
index ac20c5c..6bff6b3 100644
--- mythtv/libs/libmythbase/mythtimezone.cpp
+++ mythtv/libs/libmythbase/mythtimezone.cpp
@@ -1,6 +1,7 @@
 #include "mythtimezone.h"
 
 #include <stdlib.h> // for malloc, getenv
+#include <time.h>
 
 #include <QDataStream>
 #include <QTextStream>
diff --git mythtv/libs/libmythui/mythgesture.cpp mythtv/libs/libmythui/mythgesture.cpp
index 37846e4..f020f1a 100644
--- mythtv/libs/libmythui/mythgesture.cpp
+++ mythtv/libs/libmythui/mythgesture.cpp
@@ -30,6 +30,7 @@
 
 #include <cmath>
 #include <algorithm>
+#include <complex>
 
 #include <QMutex>
 #include <QMap>
diff --git mythtv/libs/libmythui/mythpainter.cpp mythtv/libs/libmythui/mythpainter.cpp
index c821a7b..29b063c 100644
--- mythtv/libs/libmythui/mythpainter.cpp
+++ mythtv/libs/libmythui/mythpainter.cpp
@@ -1,4 +1,5 @@
 #include <stdint.h>
+#include <complex>
 
 // QT headers
 #include <QRect>
diff --git mythtv/libs/libmythupnp/mmulticastsocketdevice.h mythtv/libs/libmythupnp/mmulticastsocketdevice.h
index ad793f9..4bfdd62 100644
--- mythtv/libs/libmythupnp/mmulticastsocketdevice.h
+++ mythtv/libs/libmythupnp/mmulticastsocketdevice.h
@@ -13,6 +13,10 @@
 #ifndef _MULTICAST_SOCKET_DEVICE_H_
 #define _MULTICAST_SOCKET_DEVICE_H_
 
+#ifdef __FreeBSD__
+#  include <sys/types.h>
+#endif
+
 #ifdef _WIN32
 # ifndef _MSC_VER
 #  include <ws2tcpip.h>
--
cgit v0.9.0.2-52-g1de3
